src/ocnicepost.fd/ocnicepost.F90

```diff
@@ -0,0 +1,577 @@
+  character(len=4), dimension(ndest)   :: dstgrds = (/'0p25', '0p5 ', '1p0 '/)
+
+  ! packed arrays on source grid
+  real, allocatable, dimension(:,:)   :: bilin2d, consd2d
+  real, allocatable, dimension(:,:,:) :: bilin3d
+  ! variable types
+  type(vardefs), allocatable :: b2d(:)
+  type(vardefs), allocatable :: c2d(:)
+  type(vardefs), allocatable :: b3d(:)
+
+  ! source grid fields
+  real, dimension(nxt,nyt,nlevs) :: tmp3d
+  real, dimension(nxt*nyt,nlevs) :: mask3d
+  real, dimension(nxt*nyt)       :: mask2d
+  real, dimension(nxt*nyt)       :: cosrot, sinrot, anglet
+  real, dimension(nlevs)         :: z_l
+  real, dimension(0:nlevs)       :: z_i
+
+  ! destination grid fields
+  real, allocatable, dimension(:,:)  :: dstlon, dstlat
+  real, allocatable, dimension(:,:)  :: rgmask3d
+  real, allocatable, dimension(:)    :: rgmask2d
+  ! output fields (x,y,z)
+  real, allocatable, dimension(:)     :: out1d
+  real, allocatable, dimension(:,:)   :: out2d
+  real, allocatable, dimension(:,:,:) :: out3d
+  ! packed remapped fields
+  real, allocatable, dimension(:,:)   :: rgb2d, rgc2d
+  real, allocatable, dimension(:,:,:) :: rgb3d
+
+  real(kind=8)       :: timestamp
+  character(len= 40) :: timeunit, timecal
+  character(len= 20) :: vname, vunit
+  character(len=120) :: vlong
+  character(len=4)   :: dstgrid
+
+  real    :: vfill
+  integer :: nd, nxr, nyr
+  integer :: i,j,k,n,nn,nvalid,iounit
+  integer :: rc,ncid,varid,dimid
+  integer :: nbilin2d,nbilin3d,nconsd2d
+  integer :: idimid,jdimid,kdimid,edimid,timid
+  integer :: idx1,idx2,idx3
+  logical :: do_ocnpost
+
+  namelist /ocnicepost_nml/ ftype, wgtsdir, debug
+
+  ! --------------------------------------------------------
+  ! read the name list
+  ! --------------------------------------------------------
+
+  fnml = 'ocnicepost.nml'
+  inquire (file=trim(fnml), iostat=rc)
+  if (rc /= 0) then
+     write (6, '(3a)') 'Error: input file "', trim(fnml), '" does not exist.'
+     stop
+  end if
+
+  ! Open and read Namelist file.
+  open (action='read', file=trim(fnml), iostat=rc, newunit=iounit)
+  read (nml=ocnicepost_nml, iostat=rc, unit=iounit)
+  if (rc /= 0) then
+     write (6, '(a)') 'Error: invalid Namelist format.'
+  end if
+  close (iounit)
+
+  ! initialize the source file type and variables
+  if (trim(ftype) == 'ocean') then
+     do_ocnpost = .true.
+     call ocnvars_typedefine
+  else
+     do_ocnpost = .false.
+     call icevars_typedefine
+  end if
+  input_file = trim(ftype)//'.nc'
+
+  open(newunit=logunit, file=trim(ftype)//'.post.log',form='formatted')
+  if (debug) write(logunit, '(a)')'input file: '//trim(input_file)
+
+  ! --------------------------------------------------------
+  ! read the source file and obtain the units and long name,
+  ! rotation angles, vertical grid and time axis
+  ! --------------------------------------------------------
+
+  nvalid = 0
+  rc = nf90_open(trim(input_file), nf90_nowrite, ncid)
```

Comment:
Can we please wrap error checking so that the netCDF calls here and below terminate the program in the event of an error?

An example is:
```fortran
subroutine nc_check(status)
  integer, intent ( in) :: status
  
    if(status /= nf90_noerr) then 
      write(6, '(a)') 'FATAL ERROR: ' // trim(nf90_strerror(status))
      stop status
    end if
end subroutine nc_check
```

and then
```fortran
call nc_check(nf90_open(trim(input_file), nf90_nowrite, ncid))
```

Reply:
I personally do not prefer this style of checking, but I can use it if preferred here.

Reply:
Understood, but we need to be able to check the return code `rc` to be non-zero and exit if it is.
I was merely providing a typical example used in a lot of our codes so it feels familiar. 
 However, feel free to use whatever you prefer.